pennylane/tape/qscript.py

```diff
@@ -865,9 +907,10 @@ def copy(self, copy_operations: bool = False) -> "QuantumScript":
         new_qscript = self.__class__(
             ops=_ops,
             measurements=_measurements,
-            shots=self.shots,
-            trainable_params=list(self.trainable_params),
+            shots=update.get("shots", self.shots),
+            trainable_params=list(update.get("trainable_params", self.trainable_params)),
         )
+
         new_qscript._graph = None if copy_operations else self._graph
         new_qscript._specs = None
         new_qscript._batch_size = self._batch_size
```

Comment:
We might not want to just immediately copy over the batch size if the operations got updated.